### PR TITLE
[preview; not ready to be merged] Wire the subnet gateway config to ENI plugin

### PR DESCRIPTION
**Pull request by aaithal**
### Summary

Wire the subnet gateway config to ENI plugin

### Implementation details

This PR complements https://github.com/aws/amazon-ecs-cni-plugins/pull/81 by wiring in the subnet gateway address in plugin args. However, we need to wait for changes in the service to be deployed before merging this. The subnet gateway ipv4 address being sent from ACS lacks netmask today, which needs to be fixed before this is merged/released.

### Testing

- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: Yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/api/eni.go [hunk 1]
```diff
@@ -41,6 +41,9 @@ type ENI struct {
 
 	// PrivateDNSName is the dns name assigned by the vpc to this eni
 	PrivateDNSName string `json:",omitempty"`
+	// SubnetGatewayIPV4Address is the address to the subnet gateway for
+	// the eni
+	SubnetGatewayIPV4Address string `json:",omitempty"`
```
Comment (ofiliz) [inline agent/api/eni.go]: Are the fields in this struct grouped by IP address family or by function? There are separate fields for IPv4Addresses and IPv6Addresses, but a single field for DomainNameServers. Should this new field be named SubnetGatewayAddress with type array consisting of both IPv4 and IPv6 gateways?

File: agent/api/eni.go [hunk 2]
```diff
@@ -68,6 +71,11 @@ func (eni *ENI) GetHostname() string {
 	return eni.PrivateDNSName
 }
 
+// GetSubnetGatewayIPV4Address returns the hostname assigned to the ENI
```
Comment (ofiliz) [inline agent/api/eni.go]: ...returns the subnet IPv4 gateway address assigned to the ENI

File: agent/ecscni/plugin.go [hunk 3]
```diff
@@ -290,13 +290,14 @@ func (client *cniClient) constructNetworkConfig(cfg interface{}, plugin string)
...
-		IPV4Address:          cfg.ENIIPV4Address,
-		MACAddress:           cfg.ENIMACAddress,
-		IPV6Address:          cfg.ENIIPV6Address,
-		BlockInstanceMetdata: cfg.BlockInstanceMetdata,
+		Type:                     ECSENIPluginName,
+		CNIVersion:               client.cniVersion,
+		ENIID:                    cfg.ENIID,
+		IPV4Address:              cfg.ENIIPV4Address,
+		MACAddress:               cfg.ENIMACAddress,
+		IPV6Address:              cfg.ENIIPV6Address,
```
Comment (ofiliz) [inline agent/ecscni/plugin.go]: Nit: Isn't the world a better place when struct field are initialized with the same order as they were defined? Now might be a good time to fix the ordering and move IPv6Address one line up :)

Comment (adnxn) [reply agent/ecscni/plugin.go]: ehh fair enough! put the IPV4 and IPV6 next to each other 😀

**Comment by adnxn** [comment]
closing in favor of https://github.com/aws/amazon-ecs-agent/pull/1491

**Pull request closed without merging**